### DISCUSSION TITLE
加减乘除那个 api 无法使用变量

### DISCUSSION THREAD

**Discussion by loveqianool** [qd-today/qd#167] [General]
某站签到需要答题，题目诸如 15 + 6，11 - 3，我把数值保存到变量，但是现在的运算 api 好像没办法使用变量呀？

希望能跟字符串替换一样输入三个参数，并且可以使用变量。
或者参考一下这个，我看了几个小时，实在是看不懂。
https://segmentfault.com/a/1190000023898398

Firefox 上无法编辑模板，不兼容 Firefox 浏览器。

**Comment by a76yyyy** [comment]
![image](https://user-images.githubusercontent.com/56478790/147483095-05f5ba97-079b-4810-a62e-e6d040c7778a.png)
加减乘除有专用的函数

**Reply by loveqianool** [reply]
就是这个函数我里面数字填变量的话它没办法计算呀？

**Comment by a76yyyy** [comment]
是最新版本吧，是的话能否提供测试用har模板或者后台报错日志

**Reply by loveqianool** [reply]
我向 q981331502@163.com 发了一个 har 模板。

**Reply by a76yyyy** [reply]
公共模板库有一个91tvg的模板，你可以搜一下看符不符合你验证的逻辑，你发的网站要邀请码我就不测试了

**Reply by loveqianool** [reply]
啊，我明白了，那个数字变量不用花括号括起来。
现在还有一个问题，他的题目有加法和减法两种，现在程序内可以判断之后调用相应的函数吗？
我给你发的模板带有正确的 cookie 的，你可以看一下。

**Reply by acooler15** [reply]
> 

现在程序没有相关的实现。 @a76yyyy 或许可以使用正则来判断加减乘除，[python数学公式识别_python用re正则表达式实现数学公式计算
](https://blog.csdn.net/weixin_39867208/article/details/110273276)

**Reply by a76yyyy** [reply]
做一个通用的api？

**Reply by acooler15** [reply]
> 做一个通用的api？

是的，新增一个api

**Comment by loveqianool** [comment]
是的，就是这个。

**Reply by a76yyyy** [reply]
按照需求在issue里提交一个特性说明吧

**Reply by loveqianool** [reply]
我就是提不了 issue 才发到这边的，不知道是网络问题还是怎么，我一点提交就会显示页面出现问题。

**Reply by a76yyyy** [reply]
有人给我反馈了这个问题，现在应该已经修复了，你可以再试试
